bokehjs/src/lib/models/transforms/jitter.ts

```diff
@@ -35,10 +36,28 @@ export class Jitter extends RangeTransform {
   }
 
   override v_compute(xs0: Arrayable<number | Factor>): Arrayable<number> {
-    if (this.previous_values != null && this.previous_values.length == xs0.length)
-      return this.previous_values
-    this.previous_values = super.v_compute(xs0)
-    return this.previous_values
+    let xs: Arrayable<number>
+    if (super.range instanceof FactorRange)
+      xs = super.range.v_synthetic(xs0)
+    else if (isArrayableOf(xs0, isNumber))
+      xs = xs0
+    else
+      throw new Error("unexpected")
+
+    if (this.previous_offsets != null && this.previous_offsets.length == xs.length) {
+      const result = new (infer_type(xs))(xs.length)
+      for (let i = 0; i < xs.length; i++) {
+        result[i] = this.previous_offsets[i] + xs[i]
+      }
+      return result
+    }
+
+    const computed_values = super.v_compute(xs)
+    this.previous_offsets = new (infer_type(xs))(xs.length)
+    for (let i = 0; i < xs.length; i++) {
+      this.previous_offsets[i] = computed_values[i] - xs[i]
```

Comment:
Calling super like this will result in double synthetic encoding in the case of of `FactorRange`. This will, however, work because synthetic coordinates are passed through `FactorRange.v_compute()`. Instead drop `super.v_compute()` and use `this._compute(xs[i])` directly. Then by using `map()`, you would get something like:
```ts
this.previous_offsets = map(xs, (xsi) => this._compute(xsi) - xsi)
````

Reply:
Changed to using `this._compute` and using `map`. Thanks for the explanation.